lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java

```
@@ -255,6 +255,9 @@ && isIntegerOrDecimalPrimitive(primitiveType)) {
                 }
```

there is still a kill switch, right?

---

yes, the flag `parquet.optimized-reader.enabled` is still there, this just avoids silent fallbacks to old reader because now all the types are expected to be supported.